packages/backend/src/handlers/triggers/updateDiscordRole.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-console */
+import Discord, { Role } from 'discord.js';
+
+import { CONFIG } from '../../config';
+import { AccountType_Enum, Player, PlayerRank_Enum } from '../../lib/autogen/hasura-sdk';
+import { client } from '../../lib/hasuraClient';
+import { TriggerPayload } from './types';
+
+export interface UpdateRole {
+  playerId: string;
+  previousRole: string | undefined;
+  newRole: string;
+}
+
+export const updateDiscordRole = async (
+  payload: TriggerPayload<Player>,
+) => {
+  const {old: oldPlayer, new: newPlayer} = payload.event.data;
+
+  try {
+    if (newPlayer == null) return;
+   
+    const getPlayerResponse = await client.GetPlayer({playerId: newPlayer.id});
+    const discordPlayerAccount = getPlayerResponse.player_by_pk?.Accounts?.find(a => a.type === AccountType_Enum.Discord);
+    if (discordPlayerAccount?.identifier == null) return;
+
+    const newRank = newPlayer?.rank;
+
+    if (newRank == null) return;
+   
+    // look up guild by guildname = 'metagame' (for now), 
+    const getGuildResponse = await client.GetGuild({guildname: 'metafam'});
+    const discordAccount = getGuildResponse.guild[0]?.guild_accounts?.find(a => a.type === AccountType_Enum.Discord);
+    if (discordAccount == null) return;
+
+    // call discord API. We'll need serverId, playerId, and roleIds
+    const discordClient = new Discord.Client();
+    discordClient.login(CONFIG.discordBotToken);
+
+    // todo add jsonb field to guild and populate ranks, e.g. { ranks: [] }
+    const guild = await discordClient.guilds.fetch(discordAccount.identifier);
```

Comment:
`discordAccount` makes it sound like it's a user account for discord, but seems like this is actually a `discordGuild`.

Reply:
renamed this.